cmd/k8s-operator/sts.go

```diff
@@ -902,6 +902,10 @@ func enableEndpoints(ss *appsv1.StatefulSet, metrics, debug bool) {
 	}
 }
 
+func isMainContainer(c *corev1.Container) bool {
```

Comment:
This function is introduced so that the `ProxyClass` field `.spec.statefulSet.pod.tailscaleContainer` can be applied to the `k8s-proxy` container, but as discussed offline with @ChaosInTheCRD, this makes the `ProxyClass` field name a bit confusing. Is it going to be possible to seamlessly rename this field to `proxyContainer`? I'm hoping so, but will save that for a follow-up PR to give it the care and attention it needs as it's quite a risky change.

Reply:
I feel like we could have kept the `tailscale` name for the new container too.

`k8s-proxy` makes sense as an image name to distinguish this from other images that we have like `tailscale/tailscale`, but to a person looking at a Kubernetes StatefulSet, it is probably more interesting to find out which container runs Tailscale - they already know that this is on k8s and it's a proxy

Reply:
Another thing that I was thinking was that there is some inconsistency:

- an image for ingress, egress ProxyGroups can be set via ProxyClass or globally via these [helm chart values](https://github.com/tailscale/tailscale/blob/47f431b656d0c35aac6f97530a4daa2404bc12d6/cmd/k8s-operator/deploy/chart/values.yaml#L95-L100)

- an image for kube-apiserver ProxyGroup _can_ be set via the ProxyClass fields, but not via the helm chart values

I don't think there is anything we can do about it apart from documenting though

Reply:
Also, I imagine that folks will often mix up the two images (I just did and was confused when the proxy was unable to get its Secret), if there was any way to easily catch some of those issues (you have configured `tailscale/tailscale` image, this will not work as the new kube api server proxy image) that could be helpful